runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -275,6 +275,21 @@ void registerTable(@Nonnull String namespace,
         }
     }
 
+    /**
+     * Check if the table's schema options changed or not. We should update the registry
+     * table when the schema options changed even if there is already an entry for the
+     * current table.
+     *
+     * @param oldOptions  - Previous schema options of the current table if it was opened before.
+     * @param newOptions  - Schema options of the table that is being registered.
+     * @return true if old schema options do not equal to new schema options. Otherwise, return false.
+     */
+    private boolean schemaOptionsChanged(@Nonnull CorfuOptions.SchemaOptions oldOptions,
+                                         @Nonnull CorfuOptions.SchemaOptions newOptions) {
+        return oldOptions.equals(newOptions);
```

Comment:
SchemaOptions is a protobuf derived Java object. The equals method is not reliable because all it does under the hood is examine the hash code of each of the child fields. There are issues with different types used for example you can check out this issue
https://github.com/protocolbuffers/protobuf/issues/2116

For this specific usage in SchemaOptions this equals() **might** work but can we try to write our **own** comparator here?

Also this only compares the schema options right, can we compare the other fields in the CorfuRecord itself?
Like @pankti-m mentioned in a comment earlier since we are doing this to support build to build testing, it is conceivable that someone modifies the key, payload and metadata types themselves and we miss updating the record to reflect the changes?